File: Search Algorithm/JumpSearch.java
import java.lang.Math; 

public class Main
{
    public static void jumpsort(int arr[], int search, int p )
    {
    int k = p-1;
    if(k<0 || k>9)
    {
        System.out.println("Value not find in array");
    }
    else if(arr[k]==search)
    {
        System.out.println("Searching element find at possition "+(k+1));
    }
    else if (arr[k]>search)
    {
        jumpsort(arr, search ,p+3);
    }
    else if(arr[k]<search)
    {
        for(int i=k-p;i<k;i++)
        {
        if(arr[i]==search)
        {
        System.out.println("Searching element find at possition"+(i+1));
        }
        }
    }
    }
    
	public static void main(String[] args) {
	int arr[] = {1,2,3,4,5,6,7,8,9,10};
	int search = 2;
	//need array in sorted  form for jump algo to work
	int p = (int)Math.sqrt(10);
	jumpsort(arr, search,p);
	
	}
}

File: JumpSearch.py
import math
def jump(arr, item):
    l = len(arr)
    jp = math.sqrt(l)
    a = 0
    while(arr[int(min(jp,l)-1)]<item):
        a = jp
        jp=jp+math.sqrt(l)
        if(a>=l):
            return -1
    while(arr[int(a)]<item):
        a=a+1
        if(a==min(jp,l)):
            return -1
    if arr[int(a)]==item:
        return a
    return -1

#the array should be sorted
arr = [11,22,33,44,52,67,72,89]
item = 89
index =jump(arr, item)
print("%.0f"%index)
